src/NuGet.Core/NuGet.Commands/PackagesLockFileBuilder.cs

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Linq;
+using NuGet.Common;
+using NuGet.LibraryModel;
+using NuGet.Packaging.Core;
+using NuGet.ProjectModel;
+using NuGet.Shared;
+
+namespace NuGet.Commands
+{
+    public class PackagesLockFileBuilder
+    {
+        private readonly int _lockFileVersion;
+
+        public PackagesLockFileBuilder(int lockFileVersion)
+        {
+            _lockFileVersion = lockFileVersion;
+        }
+
+        public PackagesLockFile CreateNuGetLockFile(LockFile assetsFile)
+        {
+            var lockFile = new PackagesLockFile()
+            {
+                Version = _lockFileVersion
+            };
+
+            var libraryLookup = assetsFile.Libraries.Where(e => e.Type == LibraryType.Package)
+                .ToDictionary(e => new PackageIdentity(e.Name, e.Version));
+
+            foreach (var target in assetsFile.Targets)
+            {
+                var nuGettarget = new PackagesLockFileTarget()
+                {
+                    TargetFramework = target.TargetFramework,
+                    RuntimeIdentifier = target.RuntimeIdentifier
+                };
+
+                var framework = assetsFile.PackageSpec.TargetFrameworks.FirstOrDefault(
+                    f => EqualityUtility.EqualsWithNullCheck(f.FrameworkName, target.TargetFramework));
+
+                var libraries = target.Libraries;
+
+                // check if this is RID-based graph then only add those libraries which differ from original TFM.
+                if (!string.IsNullOrEmpty(target.RuntimeIdentifier))
+                {
+                    var onlyTFM = assetsFile.Targets.First(t => EqualityUtility.EqualsWithNullCheck(t.TargetFramework, target.TargetFramework));
+
+                    libraries = target.Libraries.Where(lib => !onlyTFM.Libraries.Any(tfmLib => tfmLib.Equals(lib))).ToList();
+                }
+
+                foreach (var library in libraries.Where(e => e.Type == LibraryType.Package))
+                {
+                    var identity = new PackageIdentity(library.Name, library.Version);
+
+                    var dependency = new LockFileDependency()
+                    {
+                        Id = library.Name,
+                        ResolvedVersion = library.Version,
+                        Sha512 = libraryLookup[identity].Sha512,
+                        Dependencies = library.Dependencies
+                    };
+
+                    var framework_dep = framework?.Dependencies.FirstOrDefault(
+                        dep => PathUtility.GetStringComparerBasedOnOS().Equals(dep.Name, library.Name));
```

Comment:
should this be case insensitive?

Reply:
If you look at where this is coming from, it's from LibraryRange. 
LibraryRange equals compares the name as case insensitive.

Reply:
As i mentioned above, package name is case sensitive for linux and when we're enabled cross platforms we should take care of that. Currently we dont have consistent code across classes which is why we end up in comparison issues as well. But most people don't create multiple packages only difference in casings

Reply:
Just checked the origin, and indeed it's case insensitive.

Reply:
👍